sdk/appcenter-auth/src/main/java/com/microsoft/appcenter/auth/Auth.java

```diff
@@ -147,6 +155,45 @@ public void onTokenRequiresRefresh(String homeAccountId) {
         }
     };
 
+    /**
+     * Init.
+     */
+    private Auth() {
+
+        /* Setup MSAL Logging. */
+        Logger.getInstance().setLogLevel(Logger.LogLevel.VERBOSE);
+
+        try {
+            Logger.getInstance().setExternalLogger(new ILoggerCallback() {
+                @Override
+                public void log(String tag, Logger.LogLevel logLevel, String message, boolean containsPII) {
+                    if (!containsPII) {
+                        String prefixedTag = LOG_TAG + TAG_DELIMITER + tag;
+                        switch (logLevel) {
+                            case VERBOSE:
+                                AppCenterLog.verbose(prefixedTag, message);
+                                break;
+                            case INFO:
+                                AppCenterLog.info(prefixedTag, message);
+                                break;
+                            case WARNING:
+                                AppCenterLog.warn(prefixedTag, message);
+                                break;
+                            case ERROR:
+                                AppCenterLog.error(prefixedTag, message);
+                                break;
+                        }
+                    }
+                }
+            });
+        } catch (Exception e) {
+
+            /* Should only happen in tests when resetting the external logger. */
+            AppCenterLog.error(LOG_TAG, "Enabling MSAL logging failed.", e);
```

Comment:
Unit tests reset the Auth instance which in turns resets the MSAL external logger which generates the exception. Added the try and catch to prevent the tests to fail in such case.

Reply:
I am opened to better ideas.